test-result-summary-client/src/PerfCompare/PerfCompare.jsx

```diff
@@ -401,65 +442,99 @@ export default class PerfCompare extends Component {
+            curVariantData["variant"] = parsedVariantsBaseline.variant;
+            curVariantData["baselineProduct"] = parsedVariantsBaseline.product;
+            curVariantData["testProduct"] = parsedVariantsTest.product;
             curVariantData["summary"] = "";
+            curVariantData["baselineMachine"] = parsedVariantsBaseline.machine;
+            curVariantData["testMachine"] = parsedVariantsTest.machine;
 
             curMetricTable = [];
-            for (let j = 0; j < this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].metrics.length; j++) {
+            for (let j = 0; j < parsedVariantsBaseline.metrics.length; j++) {
                 
                 // Must match the baseline metric name
-                curMatchingTestMetricIndex = this.state.benchmarkRuns.benchmarkRunTest.parsedVariants[curMatchingTestVariantIndex].metrics.map(x =>
-                    x.name).indexOf(this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].metrics[j].name);
+                curMatchingTestMetricIndex = parsedVariantsTest.metrics.map(x =>
+                    x.name).indexOf(parsedVariantsBaseline.metrics[j].name);
 
                 // Metric was not found in Test run, skip it
                 if (curMatchingTestMetricIndex === -1) {
                     continue;
                 }
 
-                curMetricName = this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].metrics[j].name;
-
-                curRawValues = {
-                    "baseline": this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].metrics[j].value,
-                    "test": this.state.benchmarkRuns.benchmarkRunTest.parsedVariants[curMatchingTestVariantIndex].metrics[curMatchingTestMetricIndex].value
-                };
-
-                try {
-                    curBaselineScore = math.round((this.state.benchmarkRuns.benchmarkRunBaseline.parsedVariants[i].metrics[j].mean), 3);
-                } catch (roundBaselineScoreError) {
-                    curBaselineScore = "error";
-                }
-                
-                try {
-                    curTestScore = math.round((this.state.benchmarkRuns.benchmarkRunTest.parsedVariants[curMatchingTestVariantIndex].metrics[curMatchingTestMetricIndex].mean), 3);
-                } catch (roundTestScoreError) {
-                    curTestScore = "error";
+                // get the raw values for master nodes.
+                curMetricName = parsedVariantsBaseline.metrics[j].name;
+                let childBuildBaselineData = {};
+                let childBuildTestData = {};
+                if ( this.state.benchmarkRuns.childBuildsListBaseline && this.state.benchmarkRuns.childBuildsListBaseline.length > 0){
+                    for ( let x = 0; x < this.state.benchmarkRuns.childBuildsListBaseline.length; x++ ) {
+                        childBuildBaselineData["child " + (x + 1) ] = this.state.benchmarkRuns.childBuildsListBaseline[x][j].value;
```

Comment:
Could you please add a comment here for `childBuildsListBaseline[x][j]` to explain that we are getting the raw value for j metric of the x child?